modules/ssh/ssh.go

```diff
@@ -16,6 +22,52 @@ type Host struct {
 	SshKeyPair  *KeyPair
 }
 
+// ScpFileToE uploads the contents using SCP to the given host and fails the test if the connection fails.
+func ScpFileTo(t *testing.T, host Host, mode os.FileMode, remotePath, contents string) {
+	err := ScpFileToE(t, host, mode, remotePath, contents)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// ScpFileToE uploads the contents using SCP to the given host and return an error if the process fails.
+func ScpFileToE(t *testing.T, host Host, mode os.FileMode, remotePath, contents string) error {
+	authMethods, err := createAuthMethodsForHost(host)
+	if err != nil {
+		return err
+	}
+	file := path.Base(remotePath)
+	dir := path.Dir(remotePath)
+
+	hostOptions := SshConnectionOptions{
+		Username:    host.SshUserName,
+		Address:     host.Hostname,
+		Port:        22,
+		Command:     "/usr/bin/scp -t " + dir,
```

Comment:
Are you relying on `scp` being installed instead of implementing the file copying functionality in Go?

Reply:
You can't implement the functionality in Go as scp must be installed to handle the SCP protocol. The only other alternative would be to chain a large number of commands to `echo 'foo' >> file` and hope to not go over the maximum command limit.

Reply:
Oh, I gotcha. Despite having used it many times, I didn't actually realize that's how SCP worked. TIL! 😄